src/AutoRest.CSharp/Common/AutoRest/Plugins/Configuration.cs

```diff
@@ -205,6 +211,61 @@ public static void Initialize(IPluginCommunication autoRest)
             );
         }
 
+        private static string DownloadLatestContract(string projectFolder)
```

Comment:
To explain the change in this file:
1. When the `existingProjectFolder` is not empty, it means we are in the test.
2. When the `existingProjectFolder` is empty, it means we are in the product so we try to call `Download-Latest-Contract.ps1` implemented in https://github.com/Azure/azure-sdk-tools/pull/5587 and https://github.com/Azure/azure-sdk-for-net/pull/35001

In the function of `DownloadLatestContract`, we will start two processes sequentially:
1. Call `git rev-parse --show-toplevel` to get the root folder of current repo, so that we could know the script path for the next step. (Maybe this could be changed to just C# code, but this is not the biggest challenge I want to raise)
2. Call `Download-Latest-Contract.ps1` to download the existing contract. What's more, the script will print the project folder which contains the existing contract to stdout, so that I redirect the stdout to the pipe between the generator process and pwsh process and get the project folder from the pipe.

Everything goes well in standalone until I found it hangs as an autorest plugin. Not sure if it is because the generator and autorest has connection based on stdin and stdout, but the process will hang at `process.StandardOutput.ReadToEnd();`

Current workaround I could think of is:
1. In the `Download-Latest-Contract.ps1`, we print the project folder (containing existing contract) to a temp file.
2. In the generator, we don't redirect stdout. We just call the script `Download-Latest-Contract.ps1` and read data from the temp file generated above. After that, we delete temp file.